src/components/Carousel/Carousel.jsx

```diff
@@ -0,0 +1,239 @@
+          left: 20px;
+        `
+      : css`
+          right: 20px;
+        `}
+`;
+
+const MainList = styled.ul`
+  display: flex;
+  padding: 0;
+  margin: 0;
+  position: relative;
+  left: ${props => `${props.left}px`};
+  transition: left 0.5s ease-in;
+`;
+
+const MainListItem = styled.li`
+  list-style: none;
+`;
+
+// grabbed these styles from https://a11yproject.com/posts/how-to-hide-content/
+const Announcement = styled.div`
+  position: absolute !important;
+  height: 1px;
+  width: 1px;
+  overflow: hidden;
+  clip: rect(1px 1px 1px 1px); /* IE6, IE7 */
+  clip: rect(1px, 1px, 1px, 1px);
+  white-space: nowrap; /* added line */
+`;
+
+const IndicatorList = styled.ol`
+  padding: 0;
+  margin: 0;
+  display: flex;
+  justify-content: center;
+`;
+
+const IndicatorListItem = styled.li`
+  list-style: none;
+  padding: 0 4px;
+`;
+
+const IndicatorButton = styled.button`
+  width: 13px;
+  height: 13px;
+  border: none;
+  border-radius: 50%;
+  background-color: ${props => (props.active ? white : gray)};
+
+  :hover {
+    cursor: pointer;
+  }
+`;
+
+const Content = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+class Carousel extends React.Component {
+  state = {
+    left: 0,
+    currentSlideIndex: 0,
+  };
+
+  moveBackward = () => {
+    const { left } = this.state;
+    const { width } = this.props;
+    if (left === 0) {
+      // if it's the first item we still want to update the currentSlideIndex so we can
+      // update the aria-hidden attribute appropriately
+      this.setState(prevState => ({
+        currentSlideIndex:
+          prevState.currentSlideIndex === 0
+            ? prevState.currentSlideIndex
+            : prevState.currentSlideIndex - 1,
+      }));
+      return;
+    }
+    this.setState(prevState => ({
+      left: prevState.left + parseInt(width, 10),
+      currentSlideIndex: prevState.currentSlideIndex - 1,
+    }));
+  };
+
+  moveForward = () => {
+    const { children, width } = this.props;
+    if (this.verifyLastItem(children.length, width)) {
+      // if it's the last item we still want to update the currentSlideIndex so we can
+      // update the aria-hidden attribute appropriately
+      this.setState(prevState => ({
+        currentSlideIndex:
+          prevState.currentSlideIndex === children.length - 1
+            ? prevState.currentSlideIndex
+            : prevState.currentSlideIndex + 1,
+      }));
+      return;
+    }
+
+    this.setState(prevState => ({
+      left: prevState.left - parseInt(width, 10),
+      currentSlideIndex: prevState.currentSlideIndex + 1,
+    }));
+  };
+
+  verifyLastItem = (lengthOfChildren, widthOfEachItem) => {
+    const { left } = this.state;
+    const finalLength = (lengthOfChildren - 1) * parseInt(widthOfEachItem, 10);
+
+    return left === -finalLength;
+  };
+
+  renderItems = () => {
+    const { currentSlideIndex } = this.state;
+    const { children } = this.props;
+    const items = children.map((child, index) => (
```

Comment:
It might be worth using the helper `React.children.map(...)` to handle this map function instead of using `children.map`, because you might only pass in 1 child element, and then it won't be mappable so it could cause you errors

Reply:
great catch @karelm 🙌